File: RedSheet.DbEntities/Models/TemplateGroup.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("TemplateGroups", Schema = "dbo")]
	[Serializable]
	public partial class TemplateGroup  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key] 
	[Required]
	public Guid TemplateGroupId { get; set; } 
 
	[Required] 
	[MaxLength(500)]
	public string TemplateGroupName { get; set; } 
	[InverseProperty("TemplateGroup")]
	public ICollection<Project> Projects { get; set; } 


	public TemplateGroup()
	{
	this.Projects = new HashSet<Project>();

    }

    }
}

File: redsheet-portal/src/app/database-models/template-group.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { Project,  vTemplateGroupRecord  } from './'
export class TemplateGroup {
    constructor(templateGroup?: TemplateGroup  | vTemplateGroupRecord )  {
        let properties = [ "templateGroupId", "templateGroupName", "projects",];
        for (let property of properties)
            if (templateGroup && templateGroup[property])
                this[property] = templateGroup[property];
    }
 
    @required()
	templateGroupId : string;
 
    @required()
    @maxLength(500)
	templateGroupName : string =   undefined;
	@nested(Project)
	projects: Project[];



}
